programs/cleos/main.cpp

```diff
@@ -315,6 +318,42 @@ bandwidth_providers get_bandwidth_providers(const vector<string>& providers) {
    return bandwidthProviders;
 }
 
+struct provide_ram_params {
+    provide_ram_params(const string& providers) {
+        vector<string> pieces;
+        split(pieces, providers, boost::algorithm::is_any_of("/"));
+        if (pieces.size() != 3) {
+           std::cerr << localized("Ram provider ${p} not in 'account/provider/list,of,contracts[@permission]' form", ("p", providers)) << std::endl;
```

Comment:
Maybe it's better pass information as `account/provider[@permission]/list,of,contracts`?
This is a more natural form of representation, since `permission` refers to `provider`. And suddenly write it after the list of contracts